test_kcidb.py

```diff
@@ -429,3 +429,139 @@ def test_notify_main():
+                              "Subject: Test build: .*\x00"
+                              "Subject: Test test: .*\x00"
+                              "Subject: Test issue: .*\x00"
+                              "Subject: Test incident: .*\x00")
+
+    two_of_everything = json.dumps(dict(
+        version=dict(major=SCHEMA.major, minor=SCHEMA.minor),
+        checkouts=[
+            dict(id="test:checkout:1",
+                 git_commit_hash=git_commit_hash1,
+                 patchset_hash="",
+                 origin="test"),
+            dict(id="test:checkout:2",
+                 git_commit_hash=git_commit_hash2,
+                 patchset_hash="",
+                 origin="test"),
+        ],
+        builds=[
+            dict(id="test:build:1",
+                 origin="test",
+                 checkout_id="test:checkout:1",
+                 valid=False),
+            dict(id="test:build:2",
+                 origin="test",
+                 checkout_id="test:checkout:2",
+                 valid=False),
+        ],
+        tests=[
+            dict(id="test:test:1",
+                 origin="test",
+                 build_id="test:build:1",
+                 waived=False,
+                 status="PASS"),
+            dict(id="test:test:2",
+                 origin="test",
+                 build_id="test:build:2",
+                 waived=False,
+                 status="FAIL"),
+        ],
+        issues=[
+            dict(id="test:1",
+                 version=1,
+                 origin="test",
+                 report_url="https://test.com/bug/1",
+                 report_subject="Bug in kernel",
+                 culprit=dict(code=True)),
+            dict(id="non_test:1",
+                 version=1,
+                 origin="non_test",
+                 report_url="https://non-test.com/bug/1",
+                 report_subject="Another bug in kernel",
+                 culprit=dict(tool=True))
+        ],
+        incidents=[
+            dict(id="test:1",
+                 issue_id="test:1",
+                 issue_version=1,
+                 origin="test",
+                 test_id="test:1",
+                 present=True),
+            dict(id="non_test:1",
+                 issue_id="non_test:1",
+                 issue_version=1,
+                 origin="non_test",
+                 test_id="non_test:1",
+                 present=True)
+        ]
+    ))
+
+    assert_executes(two_of_everything,
+                    "kcidb.ingest_main",
+                    stdout_re="Subject: Test revision: .*\x00"
+                              "Subject: Test revision: .*\x00"
+                              "Subject: Test checkout: .*\x00"
+                              "Subject: Test checkout: .*\x00"
+                              "Subject: Test build: .*\x00"
+                              "Subject: Test build: .*\x00"
+                              "Subject: Test issue: .*\x00"
```

Comment:
@spbnick I get failure assertion when I have duplicate `Subject: Test issue: .*\x00`, and `Subject: Test incident: .*\x00`. It should be 2 for each one right?

Reply:
What's the actual output you get from kcidb-ingest in this case?

Reply:
![image](https://github.com/kernelci/kcidb/assets/10912671/5bc236b4-28a2-4cc3-8e80-1dc6a11bbb93)

Reply:
Sorry, screenshots are generally useless for lots of text, and these don't really include any relevant information. Can you upload the complete output of a test run to e.g. a [gist](https://gist.github.com/)?

Reply:
@spbnick , yeah I upload gist here:
https://gist.github.com/danghai/20fc34fc815f8ed5e3f33d5177468192

Reply:
Thanks! In general, text pastes from the terminal are preferable to screenshots.